File: llm_generated_code/Gemini/Python/Level1/Q7.py
def has_cycle(head):
  """
  Detects a cycle in a linked list using the Floyd's Cycle-Finding Algorithm.

  Args:
      head: The head node of the linked list.

  Returns:
      True if a cycle exists, False otherwise.
  """

  slow = fast = head
  while fast and fast.next:
    slow = slow.next
    fast = fast.next.next
    if slow == fast:
      return True
  return False

File: llm_generated_code/Gemini/CPP/Level1/Q7.cpp
#include <iostream>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

bool hasCycle(ListNode *head) {
    if (head == nullptr || head->next == nullptr) {
        return false;
    }

    ListNode *slow = head, *fast = head->next;

    while (slow != fast) {
        if (fast == nullptr || fast->next == nullptr) {
            return false;
        }
        slow = slow->next;
        fast = fast->next->next;
    }

    return true;
}

int main() {
    // Create a linked list with a cycle
    ListNode *head = new ListNode(1);
    head->next = new ListNode(2);
    head->next->next = new ListNode(3);
    head->next->next->next = new ListNode(4);
    head->next->next->next->next = head->next; // Create a cycle

    bool hasCyclePresent = hasCycle(head);
    cout << "Cycle is present: " << hasCyclePresent << endl;

    return 0;
}
